src/core/scope.ml

```diff
@@ -246,13 +246,15 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
         (* Create a new metavariable of type [TYPE] for the missing domain. *)
         let vs = Env.to_tbox env in
         _Meta (fresh_meta (Env.to_prod env _Type) (Array.length vs)) vs
-  (* Scoping of a binder (abstraction or product). *)
-  and scope_binder cons env xs t =
+  (* Scoping of a binder (abstraction, product or let-binding with [llet] to
+     true). *)
+  and scope_binder ?(llet=false) cons env xs t =
```

Comment:
Why `llet`? I would rather use name `is_let`.

I'm not sure if it is useful to have an optional argument here, this function is only used two or three times. But I let that up to you.

Reply:
It is just a matter of precise warning messages